src/redraw_scheduler.rs

```diff
@@ -4,12 +4,41 @@ use std::time::Instant;
 
 use log::trace;
 
-use crate::settings::SETTINGS;
+use crate::settings::{SETTINGS, Value};
 
 lazy_static! {
     pub static ref REDRAW_SCHEDULER: RedrawScheduler = RedrawScheduler::new();
 }
 
+#[derive(Clone)]
+struct RedrawSettings {
+    extra_buffer_frames: u64,
+}
+
+fn parse_changed_setting(name: &str, value: Option<Value>) -> Value {
+    match name {
+        "extra_buffer_frames" => {
+            let mut settings = SETTINGS.get::<RedrawSettings>();
+            if let Some(value) = value {
+                settings.extra_buffer_frames = value.as_u64().unwrap(); // TODO -- handle wrong data type

```

Comment:
I thought about this some more. I suspect for most cases, there are a finite number of basic types that we store as top level variables in vim: u64, string, int representing u64, etc

Given that thats the case, I feel like creating a generic trait which we can implement for the base types would be a better solution than creating a macro for this code. What do you think? It seems to make sense to me that macros would be a last resort when no other solution exists.